test/linalg/test_linalg_inverse.fypp

```diff
@@ -0,0 +1,210 @@
+#:include "common.fypp"
+#:set RC_KINDS_TYPES = REAL_KINDS_TYPES + CMPLX_KINDS_TYPES
+! Test inverse matrix operator
+module test_linalg_inverse
+    use testdrive, only: error_type, check, new_unittest, unittest_type
+    use stdlib_linalg_constants
+    use stdlib_linalg, only: inv,invert,operator(.inv.),eye
+    use stdlib_linalg_state, only: linalg_state_type,LINALG_ERROR
+
+    implicit none (type,external)
+    private
+    
+    public :: test_inverse_matrix
+
+    contains
+
+    !> Matrix inversion tests
+    subroutine test_inverse_matrix(tests)
+        !> Collection of tests
+        type(unittest_type), allocatable, intent(out) :: tests(:)
+        
+        allocate(tests(0))
+
+        #:for rk,rt,ri in RC_KINDS_TYPES
+        #:if rk!="xdp"
+        tests = [tests,new_unittest("${ri}$_eye_inverse",test_${ri}$_eye_inverse), &
```

Comment:
I propose to add a test for random SPD matrices. In my own implementation I have something like that:
```fortran 

 do i = 2, ndim
  orimat = 0
  call random_number(mat(1:i, 1:i))
  orimat(1:i, 1:i) = 0.5_real64 * matmul(mat(1:i, 1:i), transpose(mat(1:i, 1:)))
  mat = orimat

  call inv(mat)

  !identity matrix
  identity(1:i, 1:i) = reshape([(merge(1._real64, 0._real64, j/i.eq.mod(j,i)), j = 0, i**2 - 1)], [i, i])

  mat(1:i, 1:i) =  matmul(mat(1:i, 1:i), orimat(1:i, 1:i))

  call check(error ,  all(abs(mat(1:i, 1:i) - identity(1:i, 1:i)) < tol_real64) &
              , 'inv_pd: wrong inverse for '//value2char(i))
  if(allocated(error))return
 enddo
```

Reply:
[80a7742](https://github.com/fortran-lang/stdlib/pull/828/commits/80a7742bafe842b66766ab77b7c2c6d6b1199661)